core/src/main/java/com/valtech/aem/saas/core/bestbets/ClientBestBetsConsumerService.java

```diff
@@ -0,0 +1,165 @@
+package com.valtech.aem.saas.core.bestbets;
+
+import com.google.gson.Gson;
+import com.valtech.aem.saas.api.bestbets.BestBet;
+import com.valtech.aem.saas.api.bestbets.BestBetPayload;
+import com.valtech.aem.saas.api.bestbets.BestBetsActionFailedException;
+import com.valtech.aem.saas.api.bestbets.BestBetsConsumerService;
+import com.valtech.aem.saas.core.http.client.SearchRequestExecutorService;
+import com.valtech.aem.saas.core.http.request.SearchRequest;
+import com.valtech.aem.saas.core.http.request.SearchRequestDelete;
+import com.valtech.aem.saas.core.http.request.SearchRequestGet;
+import com.valtech.aem.saas.core.http.request.SearchRequestPost;
+import com.valtech.aem.saas.core.http.request.SearchRequestPut;
+import com.valtech.aem.saas.core.http.response.BestBetsDataExtractionStrategy;
+import com.valtech.aem.saas.core.http.response.ModifiedBestBetIdExtractionStrategy;
+import com.valtech.aem.saas.core.http.response.SearchResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import lombok.Builder;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.entity.EntityBuilder;
+import org.apache.http.entity.ContentType;
+
+@Builder
+public class ClientBestBetsConsumerService implements BestBetsConsumerService {
+
+  private final SearchRequestExecutorService searchRequestExecutorService;
+  private final String commonPath;
+  private final String addBestBetAction;
+  private final String addBestBetsAction;
+  private final String getBestBetsAction;
+  private final String updateBestBetAction;
+  private final String deleteBestBetAction;
+  private final String publishBestBetsForProjectAction;
+
+  @Override
+  public void addBestBet(@NonNull BestBetPayload bestBetPayload) {
+    validateCommonConfigs();
+    if (StringUtils.isBlank(addBestBetAction)) {
```

Comment:
This case seems a bit trivial. IMO extracting to a method is not really needed. If It is needed to be extracted, I’d do it in a helper class, because this type of validations is performed in other classes as well. Wdyt?

Reply:
imo helper class would be too much for this small function, but from other side i think if we leave it like this, sonar will complain for duplicate code. If thats not the case, i would say leave it like this

Reply:
Neither sonar lint plugin nor sonar cloud analysis job have reported it.